backend/src/swa-tracker/swa-tracker-wrapper.ts

```diff
@@ -0,0 +1,108 @@
+ * This enables concise and simple consumption of the swaTracker throughout our Extension.
+ *
+ * @returns { SWATracker }
+ */
+export function getSWA(): SWATracker {
+	if (isInitialized() === false) {
+		throw Error(ERROR_ANALYTICS_TRACKER_NOT_INITIALIZED);
+	}
+	return swaTracker;
+}
+
+export function createSWATracker() {
+	const swaTracker = new SWATracker(
+		"SAPSE",
+		YEOMAN_UI,
+		// callback for error, one such callback for all the errors we receive via all the track methods
+		// error can be string (err.message) or number (response.statusCode)
+		(error: string | number) => {
+			if (typeof error === 'string') {
+				getLogger().error("SAP Web Analytics tracker failed to track", {errorMessage: error});
+			} else if (typeof error === 'number') {
+				// bug in matomo-tracker: they think that success is 200 or 30[12478], so we are rechecking here
+				if (!((error >= 200) && (error <= 299)) && !((error >= 300) && (error <= 399))) {
+					getLogger().error("SAP Web Analytics tracker failed to track", {statusCode: error});
+				}
+			}
+		}
+	  );
+
+	// Update the swa-tracker-wrapper with a reference to the swaTracker.
+	initSWATracker(swaTracker);
+	getLogger().info(`SAP Web Analytics tracker was created for ${YEOMAN_UI}`);
+}
+
+export function updateGeneratorStarted(generatorName: string) {
+	const eventType = EVENT_TYPES.PROJECT_GENERATION_STARTED;
+	let customEvents = [generatorName];
+	startTime = Date.now();
+	getSWA().track(eventType, customEvents);
+	getLogger().trace("SAP Web Analytics tracker was called and start time was initialized", {
+		eventType, generatorName, startTime, customEvents});
+}
+
+export function updateGeneratorEnded(generatorName: string, isSucceeded: boolean, errorMessage?: string) {
+	let eventType = EVENT_TYPES.PROJECT_GENERATED_SUCCESSFULLY;
+	if (!isSucceeded) {
+		eventType = EVENT_TYPES.PROJECT_GENERATION_FAILED;
+	}
+
+	const endTime = Date.now();
+	const generationTimeMilliSec = endTime - startTime;
+	const generationTimeSec = Math.round(generationTimeMilliSec/1000);
+	let customEvents = [generatorName, generationTimeSec.toString()];
+	if (!_.isNil(errorMessage)) {
+		customEvents.push(errorMessage);
+	}
+	getSWA().track(eventType, customEvents);
+	getLogger().trace("SAP Web Analytics tracker was called", 
+		{eventType, generatorName, generationTimeSec, generationTimeMilliSec, endTime, startTime, customEvents, errorMessage});
+}
+
+export function updateExploreAndInstallGeneratorsLinkClicked() {
+	const eventType = EVENT_TYPES.EXPLORE_AND_INSTALL_GENERATORS_LINK;
+	getSWA().track(eventType);
+	getLogger().trace("SAP Web Analytics tracker was called", {eventType});
+}
+
+export function updateOneOfPreviousStepsClicked(generatorName: string) {
+	const eventType = EVENT_TYPES.PREVIOUS_STEP;
+	let customEvents = [generatorName];
+	getSWA().track(eventType, customEvents);
+	getLogger().trace("SAP Web Analytics tracker was called", {eventType, generatorName, customEvents});  
+}
```

Comment:
looks ok to me too.
 refactor I would do is:
export from this file only 1 class called SWA which contains all the funcs.